### PR TITLE
Added modal test fixture

### PR DESCRIPTION
**Pull request by Blackbaud-TrevorBurch**
#110

### CODE REVIEW

File: src/app/public/testing/modal-fixture.ts [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+      throw new Error(`No element was found with a \`data-sky-id\` value of "${skyTestId}".`);
+    }
+  }
+
+  /**
+   * The modal component's ARIA describedby attribute.
+   */
+  public get ariaDescribedBy(): string {
+    const modalElement = this.getModalDiaglogElement();
+    return modalElement.getAttribute('aria-describedby') || undefined;
```
Comment (Blackbaud-AlexKingman) [inline src/app/public/testing/modal-fixture.ts:28]: Should we have some null checks on `modalElement` before we try to access its attributes? The same goes for a few of the methods here.

File: src/app/public/testing/modal-fixture.ts [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+  /**
+   * The modal component's role attribute.
+   */
+  public get ariaRole(): string {
+    const modalElement = this.getModalDiaglogElement();
+    return modalElement.getAttribute('role') || undefined;
+  }
+
+  /**
+   * Whether or not the modal is a full page modal
```
Comment (Blackbaud-AlexKingman) [inline src/app/public/testing/modal-fixture.ts:48]: Minor note - some of your comments don't have punctuation while others do.

File: src/app/public/testing/modal-fixture.ts [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+   */
+  public get tiledBody(): boolean {
+    return this.getModalEl().classList.contains('sky-modal-tiled');
+  }
+
+  /**
+   * Clicks the modal header's "close" button
+   */
+  public clickHeaderCloseButton(): void {
+    const clostButton: HTMLElement = this.modalElement
```
Comment (Blackbaud-AlexKingman) [inline src/app/public/testing/modal-fixture.ts:79]: Typo: `clostButton`.

File: src/app/public/testing/modal-fixture.ts [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+      .querySelector('.sky-modal .sky-modal-btn-close');
```
Comment (Blackbaud-AlexKingman) [inline src/app/public/testing/modal-fixture.ts:80]: FYI - in modern theme, there will be no close button. Might want a null check on the returned element to throw a warning to the consumer before you run `click` on it.

File: src/app/public/testing/modal-fixture.ts [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+    clostButton.click();
+    this.fixture.detectChanges();
+  }
+
+  /**
+   * Clicks the modal header's "help" button
+   */
+  public clickHelpButton(): void {
+    const helpButton: HTMLElement = this.modalElement
+      .querySelector('.sky-modal .sky-modal-header-buttons button[name="help-button"]');
```
Comment (Blackbaud-AlexKingman) [inline src/app/public/testing/modal-fixture.ts:90]: Same here - should we throw a warning if no help button isn't found?

File: src/app/public/testing/modal-fixture.spec.ts [hunk 6]
```diff
@@ -0,0 +1,494 @@
...
+    fixture.detectChanges();
+
+    fixture.componentInstance.ariaDescribedBy = 'describingID';
+    fixture.componentInstance.launchModal();
+
+    fixture.detectChanges();
+    tick();
+    fixture.detectChanges();
+
+    const modal = new SkyModalFixture(
```
Comment (Blackbaud-AlexKingman) [inline src/app/public/testing/modal-fixture.spec.ts:205]: Could we componentize this to reduce some redundancy? Something like.. `const modal = launchModal()`? And the helper function would look something like...
```
launchModal(): SkyModalFixture {
    fixture.componentInstance.launchModal();

    fixture.detectChanges();
    tick();
    fixture.detectChanges();

    const modal = new SkyModalFixture(
      fixture,
      'test-modal'
    );

    fixture.detectChanges();

    return modal;
}
```

File: src/app/public/testing/modal-fixture.ts (hunk 2 above)
Comment (Blackbaud-TrevorBurch) [reply src/app/public/testing/modal-fixture.ts:48]: Added punctuation throughout. Let me know if I should have removed all of them :) It appeared that we normally have them in our documentation.

File: src/app/public/testing/modal-fixture.ts (hunk 1 above)
Comment (Blackbaud-TrevorBurch) [reply src/app/public/testing/modal-fixture.ts:28]: Added checks throughout.

File: src/app/public/testing/modal-fixture.ts (hunk 3 above)
Comment (Blackbaud-TrevorBurch) [reply src/app/public/testing/modal-fixture.ts:79]: Doh. Fixed

File: src/app/public/testing/modal-fixture.ts (hunk 4 above)
Comment (Blackbaud-AlexKingman) [reply src/app/public/testing/modal-fixture.ts:80]: Maybe something like this? https://github.com/blackbaud/skyux-datetime/blob/master/src/app/public/testing/datepicker-fixture.ts#L79

Comment (Blackbaud-TrevorBurch) [reply src/app/public/testing/modal-fixture.ts:95]: Done

File: src/app/public/testing/modal-fixture.ts (hunk 5 above)
Comment (Blackbaud-TrevorBurch) [reply src/app/public/testing/modal-fixture.ts:110]: Done

File: src/app/public/testing/modal-fixture.spec.ts (hunk 6 above)
Comment (Blackbaud-TrevorBurch) [reply src/app/public/testing/modal-fixture.spec.ts]: Done

File: src/app/public/testing/modal-fixture.ts (hunk 1 above)
Comment (Blackbaud-AlexKingman) [reply src/app/public/testing/modal-fixture.ts]: Do you think returning an `undefined` value is as valuable as alerting the consumer that the modal isn't found? Would it be more helpful if the test threw an error when the `modalElement` is null?
